### PR TITLE
MP-9075 NOX - Refactor EntityControllerGenerator to use Scriban template - Creating placeholder templates

### PR DESCRIPTION
**Pull request by emir-sehovic1**

### CODE REVIEW

File: src/Nox.Generator/Presentation/Api/OData/EntityController.RelationshipsGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            return;
+        }
+
+        const string templateName = @"Presentation.Api.OData.EntityController.Relationships";
+
+        foreach (var entity in codeGeneratorState.Solution.Domain.Entities)
+        {
+            context.CancellationToken.ThrowIfCancellationRequested();
+
+            if (codeGeneratorState.Solution.Domain.Entities.Any(e => e.OwnedRelationships != null && e.OwnedRelationships.Any(r => r.Entity.Equals(entity.Name))))
```
Comment (rochar) [inline src/Nox.Generator/Presentation/Api/OData/EntityController.RelationshipsGenerator.cs]: here you want the relationships and not owned

File: src/Nox.Generator/Presentation/Api/OData/EntityController.EntityGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            return;
+        }
+
+        const string templateName = @"Presentation.Api.OData.EntityController.Entity";
+
+        foreach (var entity in codeGeneratorState.Solution.Domain.Entities)
+        {
+            context.CancellationToken.ThrowIfCancellationRequested();
+
+            if (codeGeneratorState.Solution.Domain.Entities.Any(e => e.OwnedRelationships != null && e.OwnedRelationships.Any(r => r.Entity.Equals(entity.Name))))
```
Comment (rochar) [inline src/Nox.Generator/Presentation/Api/OData/EntityController.EntityGenerator.cs]: this if is not necessary

File: src/Nox.Generator/Presentation/Api/OData/EntityController.CustomQueriesGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            return;
+        }
+
+        const string templateName = @"Presentation.Api.OData.EntityController.CustomQueries";
+
+        foreach (var entity in codeGeneratorState.Solution.Domain.Entities)
+        {
+            context.CancellationToken.ThrowIfCancellationRequested();
+
+            if (codeGeneratorState.Solution.Domain.Entities.Any(e => e.OwnedRelationships != null && e.OwnedRelationships.Any(r => r.Entity.Equals(entity.Name))))
```
Comment (rochar) [inline src/Nox.Generator/Presentation/Api/OData/EntityController.CustomQueriesGenerator.cs]:
```suggestion
            if (entity.Queries== null || !entity.Queries.Any()))
```

File: src/Nox.Generator/Presentation/Api/OData/EntityController.CustomCommandsGenerator.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        }
+
+        const string templateName = @"Presentation.Api.OData.EntityController.CustomCommands";
+
+        foreach (var entity in codeGeneratorState.Solution.Domain.Entities)
+        {
+            context.CancellationToken.ThrowIfCancellationRequested();
+
+            if (codeGeneratorState.Solution.Domain.Entities.Any(e => e.OwnedRelationships != null && e.OwnedRelationships.Any(r => r.Entity.Equals(entity.Name))))
+            {
```
Comment (rochar) [inline src/Nox.Generator/Presentation/Api/OData/EntityController.CustomCommandsGenerator.cs]:
```suggestion
           if (entity.Commands== null || !entity.Commands.Any()))
```

**Review by rochar — COMMENTED** [review]
looks good, just review the comments

**Comment by emir-sehovic1** [comment]
> looks good, just review the comments

@rochar Thanks for the feedback. As per our discussion I have left the condition (just replaced it with a simpler one entity.IsOwnedEntity) as it is needed to determine whether controller is needed for the specified entity.

**Review by rochar — APPROVED** [review]

**Review by gjorgji-grgovski-authoritypartners — APPROVED** [review]

**Pull request merged**